pkg/state/state.go

```diff
@@ -354,11 +354,26 @@ func New(ctx context.Context, pgURL, stateSchema string) (*State, error) {
 }
 
 func (s *State) Init(ctx context.Context) error {
-	// ensure pgroll internal tables exist
-	// TODO: eventually use migrations for this instead of hardcoding
-	_, err := s.pgConn.ExecContext(ctx, fmt.Sprintf(sqlInit, pq.QuoteIdentifier(s.schema)))
+	tx, err := s.pgConn.Begin()
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback()
 
-	return err
+	// Try to obtain an advisory lock
+	const key int64 = 0x2c03057fb9525b
```

Comment:
Adding a comment here to explain this is a (randomly picked) unique int64 key for access to the pgroll initialisation resource that will release automatically at the end of the transaction would be good for readability/understanding when not familiar with advisory locks.

Reply:
++ also making it a constant would make sense